File: hw/quina/u2/hw19Inheritance/FarmSystemJS/ec.edu.espe.FarmSystem.view/FarmSystem.js
// Maryuri Quiña
import fs from 'fs';
import { Location } from '../ec.edu.espe.FarmSystem.model/Location.js';
import { Cage } from '../ec.edu.espe.FarmSystem.model/Cage.js';
import { FarmAnimal } from '../ec.edu.espe.FarmSystem.model/FarmAnimal.js';
import { Chicken } from '../ec.edu.espe.FarmSystem.model/Chicken.js';
import { Cow } from '../ec.edu.espe.FarmSystem.model/Cow.js';
import { Pig } from '../ec.edu.espe.FarmSystem.model/Pig.js';
import { Sheep } from '../ec.edu.espe.FarmSystem.model/Sheep.js';

function showAnimalCard(animal, number) {
  console.log('\n----------------------------------------');
  console.log(`  ANIMAL #${number}  (${animal.constructor.name})`);
  console.log('----------------------------------------');
  console.log(` Id:            ${animal.id}`);
  console.log(` Breed:         ${animal.breed}`);
  console.log(` Gender:        ${animal.gender}`);
  console.log(` Born on:       ${animal.bornOn.toISOString().slice(0, 10)}`);
  console.log(` Weight (kg):   ${animal.weight}`);
  console.log(` Reproduce:     ${animal.isAbleToReproduce}`);
  console.log(' Cage:');
  console.log(`   - Id:        ${animal.cage.id}`);
  console.log(`   - Type:      ${animal.cage.type}`);
  console.log(`   - Desc:      ${animal.cage.description}`);
  console.log(
    `   - Location:  (${animal.cage.location.xCoordinate}, ${animal.cage.location.yCoordinate})`
  );

  if (animal instanceof Chicken) {
    console.log(' Extra (Chicken):');
    console.log(`   - Is molting: ${animal.isMolting}`);
    console.log(`   - Laid eggs:  ${animal.laidEggs}`);
  } else if (animal instanceof Cow) {
    console.log(' Extra (Cow):');
    console.log(`   - Produces milk: ${animal.isProducingMilk}`);
    console.log(`   - Liters/day:    ${animal.litersADay}`);
  } else if (animal instanceof Sheep) {
    console.log(' Extra (Sheep):');
    console.log(`   - Last shearing: ${animal.lastShearing.toISOString().slice(0, 10)}`);
  } else if (animal instanceof Pig) {
    console.log(' Extra (Pig):');
    console.log('   - No extra fields');
  }

  console.log('----------------------------------------');
}

function main() {
  const lastShearing = new Date(2006, 10, 4); // mes 10 = noviembre
  const xCoordinate = 10;
  const yCoordinate = 20;
  const isAbleToReproduce = false;
  const weight = 10.4;
  const gender = 'male';
  const bornOn = new Date(2025, 1, 1); // mes 1 = febrero
  const breed = 'Holstein';

  const location = new Location(xCoordinate, yCoordinate);
  const cage = new Cage(1, 'Corral for cows', 2, location);

  /** @type {FarmAnimal[]} */
  const animals = [];

  animals.push(new Chicken(true, 0, 1, breed, bornOn, gender, isAbleToReproduce, weight, cage));
  animals.push(new Cow(true, 1.5, 2, breed, bornOn, gender, isAbleToReproduce, weight, cage));
  animals.push(new Pig(3, breed, bornOn, gender, isAbleToReproduce, weight, cage));
  animals.push(new Sheep(lastShearing, 4, breed, bornOn, gender, isAbleToReproduce, weight, cage));

  console.log('========== FARM ANIMALS REPORT ==========');
  animals.forEach((a, i) => showAnimalCard(a, i + 1));
  console.log('\n============== END OF REPORT ==============');

  // Guardar también en JSON
  const jsonData = animals.map(a => ({
    type: a.constructor.name,
    id: a.id,
    breed: a.breed,
    gender: a.gender,
    bornOn: a.bornOn.toISOString(),
    weight: a.weight,
    isAbleToReproduce: a.isAbleToReproduce,
    cage: {
      id: a.cage.id,
      type: a.cage.type,
      description: a.cage.description,
      location: {
        xCoordinate: a.cage.location.xCoordinate,
        yCoordinate: a.cage.location.yCoordinate
      }
    }
  }));

  fs.writeFileSync('animals.json', JSON.stringify(jsonData, null, 2));
  console.log('\n>> Data saved to animals.json');
}

main();

File: hw/toapanta/u2/hw19inherence/pythonFarmSystem/FarmSystem.py
<<<<<<< HEAD
from Location import Location
from Cage import Cage
from FarmAnimal import FarmAnimal
from Chicken import Chicken
from Cow import Cow
from Pig import Pig
from Sheep import Sheep
from datetime import datetime
if __name__ == "__main__":

    x_coordinate = 10
    y_coordinate = 20
    is_able_to_reproduce = False
    weight = 10.4
    gender = "male"
    
    born_on = datetime(2025, 2, 1)
    last_shearing = datetime(2025, 5, 1)
    
    breed = "Holstein"
    
    location = Location(x_coordinate, y_coordinate)
    cage = Cage(1, "Corral for cows", 2)
    
    farm_animal = Chicken(True, 0, 1, breed, born_on, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --> " + str(farm_animal))
    
    farm_animal = Cow(True, 1.5, 2, breed, born_on, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --> " + str(farm_animal))
    
    farm_animal = Sheep(last_shearing, 3, breed, born_on, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --> " + str(farm_animal))
    
    farm_animal = Pig(4, breed, born_on, gender, is_able_to_reproduce, weight, cage)
=======
from Location import Location
from Cage import Cage
from FarmAnimal import FarmAnimal
from Chicken import Chicken
from Cow import Cow
from Pig import Pig
from Sheep import Sheep
from datetime import datetime
if __name__ == "__main__":

    x_coordinate = 10
    y_coordinate = 20
    is_able_to_reproduce = False
    weight = 10.4
    gender = "male"
    
    born_on = datetime(2025, 2, 1)
    last_shearing = datetime(2025, 5, 1)
    
    breed = "Holstein"
    
    location = Location(x_coordinate, y_coordinate)
    cage = Cage(1, "Corral for cows", 2)
    
    farm_animal = Chicken(True, 0, 1, breed, born_on, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --> " + str(farm_animal))
    
    farm_animal = Cow(True, 1.5, 2, breed, born_on, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --> " + str(farm_animal))
    
    farm_animal = Sheep(last_shearing, 3, breed, born_on, gender, is_able_to_reproduce, weight, cage)
    print("farmAnimal --> " + str(farm_animal))
    
    farm_animal = Pig(4, breed, born_on, gender, is_able_to_reproduce, weight, cage)
>>>>>>> 203b676d8f105a34d549ec251a59c11aa2c57532
    print("farmAnimal --> " + str(farm_animal))